pages/resources/quizzes/FirstLang/FirstProgrammingLanguageQuiz.tsx

```diff
@@ -65,9 +59,21 @@ const FirstProgrammingLanguageQuiz = () => {
   };
 
   const handleOptionClick = (option: QuizOptionViewState) => {
-    // TODO create a new quizViewState and then call setQuizViewState
-    console.log(option, quizViewState);
-    // TODO generate a new quiz state based off the old state plus the clicked option
+    const selectedQuizOption = quizViewState.questions.map((question) => ({
+      ...question,
+      options: question.options.map((questionOption) =>
+        questionOption.name === option.name
+          ? { ...questionOption, isSelected: true }
```

Comment:
nice use of ternary and spread operator!

Reply:
how does deselction work? is that coming later?